proofs/tree.go

```diff
@@ -8,8 +8,39 @@ would not be in a javascript implementation.
 
 Note: this is a basic implementation that lacks support for serializing more complex structs. The interfaces and
 functions in this library will change significantly in the near future.
+
+Example Usage
```

Comment:
Isn't better to put a link to the example file instead of copying here the content? That way if we change the example file we do not have to change it in two places.

Reply:
Copying here makes sense because that way it shows up in the godocs online. This is something I've seen done several times.